roachtest: move the logger from cluster to test

andreimatei opened the pull request:
Before this patch, c.l was essentially a test's logger. But that logger
belongs better on a test than on a cluster, and so I've moved it.
c.l still exists, and it's being set at a high level to t.l. The idea is
that c.l will still be used, at least for the time being, by other
cluster methods, but otherwise tests will switch to using t.l instead of
c.l (this is being done in the next commit).

Release note: None

andreimatei committed:
roachtest: move the logger from cluster to test

Before this patch, c.l was essentially a test's logger. But that logger
belongs better on a test than on a cluster, and so I've moved it.
c.l still exists, and it's being set at a high level to t.l. The idea is
that c.l will still be used, at least for the time being, by other
cluster methods, but otherwise tests will switch to using t.l instead of
c.l (this is being done in the next commit).

Release note: None

andreimatei committed:
roachtest: tests switch to t.l from c.l

Now that t.l exists.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/31321)
<!-- Reviewable:end -->

andreimatei commented:
First commit is elsewhere.

andreimatei committed:
roachtest: remove the artifacts global

Recently we got better at plumbing the dirs where different log files
should go to through. This patch completes that migration by removing
the artifacts global.
A buglet is also fixed - the global was being used by the Jepsen tests
directly, in ignorance of the fact that, if --count is used, logs need
to go to a different dir for each iteration.

Release note: None

benesch reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but I didn't look at the first commit

> Reviewed 3 of 3 files at r2, 27 of 27 files at r3, 6 of 6 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-:-LOyKTqk8p0Rw78LbyMg:bcz6h9d)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/cmd/roachtest/cluster.go, line 832 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-LOyJZYr9SZ2hZU0FMBl:-LOyJZYs7Vg3QJHawyui:bvua3zf) ([raw file](https://github.com/cockroachdb/cockroach/blob/6c023e8bf86fbd6caf780ad74c7d6e515e626410/pkg/cmd/roachtest/cluster.go#L832)):*
> ```Go
> 	// This cloned cluster is not taking ownership. The parent retains it.
> 	cpy.owned = false
> 	cpy.destroyed = nil
> ```

Is this in the right commit?
___
*[pkg/cmd/roachtest/test.go, line 919 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-LOyJuLFCQNcphXCbEto:-LOyJuLFCQNcphXCbEtp:bvua3zf) ([raw file](https://github.com/cockroachdb/cockroach/blob/a564e005a2744b9c0607f1df12a10d8f32457c57/pkg/cmd/roachtest/test.go#L919)):*
> ```Go
> 			c = c.clone()
> 		}
> 		c.setTest(t)
> ```

Is this in the right commit?


<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31321)* status: :shipit: complete! 1 of 0 LGTMs obtained

___
*[pkg/cmd/roachtest/cluster.go, line 832 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-LOyJZYr9SZ2hZU0FMBl:-LOyQh14-4WZZrHriJm4:bx2stt1) ([raw file](https://github.com/cockroachdb/cockroach/blob/6c023e8bf86fbd6caf780ad74c7d6e515e626410/pkg/cmd/roachtest/cluster.go#L832)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Is this in the right commit?
</blockquote></details>

no, moved to the previous one. thanks
___
*[pkg/cmd/roachtest/test.go, line 919 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-LOyJuLFCQNcphXCbEto:-LOyM6P87ru41tTvEuag:bwqwk8p) ([raw file](https://github.com/cockroachdb/cockroach/blob/a564e005a2744b9c0607f1df12a10d8f32457c57/pkg/cmd/roachtest/test.go#L919)):*
<details><summary><i>Previously, benesch (Nikhil Benesch) wrote…</i></summary><blockquote>

Is this in the right commit?
</blockquote></details>

yes. The `setTest()` call got moved below the `clone()` call; `clone()` used to take a test, but no longer.


<!-- Sent from Reviewable.io -->

benesch reviewed (approved):
> Reviewed 31 of 31 files at r5, 26 of 26 files at r6, 6 of 6 files at r7.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31321)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

andreimatei reviewed (commented):
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/31321#-:-LOyXcp5AhBzMgQl2R0J:b-ydi9ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->
